docs/design/import.md

```diff
@@ -0,0 +1,314 @@
+# Import
+
+Practitioners may have previously existing resources that they wish to bring under Terraform's management. Conceptually, this is called importing as Terraform must create the state of the resource. Providers are responsible for receiving a request for resource import and respond with the relevant resource state(s).
+
+This design documentation will walk through and recommend options for import handling in the framework.
+
+## Background
+
+### Terraform CLI
+
+Terraform CLI supports two forms of interacting with unmanaged resources:
+
+- [`terraform add`](https://www.terraform.io/docs/cli/commands/add.html) (*Experimental*, Terraform CLI version 1.1.0 and later): Given a resource address and using the resource schema provided by the provider, generate a configuration template.
+- [`terraform import`](https://www.terraform.io/docs/cli/commands/import.html) (Terraform CLI version 0.7.0 and later): Given a resource address and (currently only) an identifier, forwards the request to the relevant provider and expects relevant resource state(s) with enough information to perform a refresh, then performs that resource refresh to ensure the resource state is fully populated.
+
+Since `add` uses already available schema information and does not require additional provider interaction, this design focuses on the latter, which is a currently unimplemented integration point for providers in the framework.
+
+### Terraform Plugin Protocol
+
+The protocol defines the following implementation details for import handling:
+
+```protobuf
+service Provider {
+    // ... other RPCs ...
+    rpc ImportResourceState(ImportResourceState.Request) returns (ImportResourceState.Response);
+}
+
+message ImportResourceState {
+    message Request {
+        string type_name = 1;
+        string id = 2;
```

Comment:
It may be helpful somewhere to call out how practitioners see this, e.g.,

```sh
$ terraform import $ADDRESS $ID
```

and to talk about what the ID is/needs to be (i.e., enough information to identify the resource or resources in the API).

Reply:
Will be included in next commit. 👍